src/routes/dashboard/containers/DashboardContainer.js

```diff
@@ -99,6 +99,10 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
     }
   }
 
+  if (route.status === 'loading') {
+    return { status: 'loading' }
+  }
```

Comment:
This seems like a hacky workaround to me. The route should only the status `loading` while there are network requests pending.

I came across this problem as well when working on the tests for the Dashboard and CategoriesContainer, and I fixed the bug there by adding a new status to the route.
I will propose a new PR that solves this (separately from the test-task)